service/controller/resource/azureclusterconfig/create.go

```diff
@@ -0,0 +1,274 @@
+package azureclusterconfig
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	releasev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/reconciliationcanceledcontext"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/pkg/label"
+	"github.com/giantswarm/azure-operator/v4/service/controller/controllercontext"
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	azureCluster, err := key.ToAzureCluster(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "finding required cluster api types")
+
+	var cluster capiv1alpha3.Cluster
+	{
+		nsName := types.NamespacedName{
+			Name:      key.ClusterName(&azureCluster),
+			Namespace: azureCluster.Namespace,
+		}
+		err = r.ctrlClient.Get(ctx, nsName, &cluster)
+		if errors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("referenced Cluster CR (%q) not found", nsName.String()))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling reconciliation")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var masterMachines []capzv1alpha3.AzureMachine
+	var workerMachines []capzv1alpha3.AzureMachine
```

Comment:
Currently this expected `AzureMachine` CRs for worker nodes that exists in the built-in worker VMSS (i.e. which is not node pool).